closeable/close.go

```diff
@@ -3,13 +3,15 @@ package closeable
 import (
 	"context"
 	"io"
+	"reflect"
 
 	"github.com/fabric8-services/fabric8-wit/log"
 )
 
 // Close closes the given resource and logs the error if something wrong happened
 func Close(ctx context.Context, c io.Closer) {
-	if c != nil {
+	// need to verify that the value of the `c` interface if not nil, too
+	if c != nil && !reflect.ValueOf(c).IsNil() {
```

Comment:
You don't actually need to check if c !=nil here. 
reflect.ValueOf(c).IsNil() is enough

Reply:
no, I tried and it panics:
````
--- FAIL: TestCloseable (0.00s)
    --- PASS: TestCloseable/non_nil (0.00s)
    --- FAIL: TestCloseable/nil (0.00s)
    	/Users/xcoulon/code/go/src/github.com/fabric8-services/fabric8-wit/closeable/assertions.go:247: 
                          
	Error Trace:	close_test.go:31
    		
	Error:      	func (assert.PanicTestFunc)(0x13f7d00) should not panic
    		
	            	
	Panic value:	reflect: call of reflect.Value.IsNil on zero Value
    		
	Test:       	TestCloseable/nil
    --- PASS: TestCloseable/non_nil_with_nil_value (0.00s)
````